### PR TITLE
Test all datatypes translation

### LINKED ISSUE #24470: Ensure all source-mysql datatypes are represented correctly in airbyte datatypes
All possible [mysql datatypes](https://dev.mysql.com/doc/refman/8.0/en/data-types.html) need to convert correctly to both airbyte datatype v0 ~and v1~.
Ensure and file issues for any work that needs to be done.

~V1 Ref: https://docs.google.com/document/d/1FupBVPMlDTU2QnEOxN42BKvdWk-Yha8604MnU5EmDRQ/edit?usp=sharing~

Airbyte protocol datatype ref: https://docs.airbyte.com/understanding-airbyte/supported-data-types
MySql types to JDBC ref: https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-type-conversions.html

### PR DESCRIPTION
**Pull request by rodireich**
## What
Test all mysql datatypes conversion behaviour.
Including precision and decimal and charset variations.
Any breaking future change should be caught with this test.

closes #24470.

### CODE REVIEW

**Review by subodh1810 — DISMISSED** [review]
This should also test in CDC mode right

**Comment by rodireich** [comment]
I didn't add initially as there are no values written to table. but since column are created and read by debezium I guess it makes sense to do the same through CDC. Added

File: airbyte-config-oss/init-oss/src/main/resources/seed/oss_catalog.json [hunk 1]
```diff
@@ -12594,11 +12594,7 @@
             "type": "string",
             "title": "Reader Options",
             "description": "This should be a string in JSON format. It depends on the chosen file format to provide additional options and tune its behavior.",
-            "examples": [ "{}", "{\"sep\": \" \"}", {
-              "sep": "\t",
-              "header": 0,
-              "names": [ "column1", "column2" ]
-            } ]
+            "examples": [ "{}", "{\"sep\": \" \"}", "{\"sep\": \"\t\", \"header\": 0, \"names\": [\"column1\", \"column2\"] }" ]
```
Comment (akashkulk) [inline airbyte-config-oss/init-oss/src/main/resources/seed/oss_catalog.json:12597]: Is this mainly a formatting change?

**Review by akashkulk — APPROVED** [review]

File: airbyte-config-oss/init-oss/src/main/resources/seed/oss_catalog.json (hunk 1 above)
Comment (rodireich) [reply airbyte-config-oss/init-oss/src/main/resources/seed/oss_catalog.json:12597]: This came with some GitHub changes. not my change. 
I'll just revert it

File: airbyte-config/init/src/main/resources/seed/oss_catalog.json
Comment (rodireich) [inline airbyte-config/init/src/main/resources/seed/oss_catalog.json:1]: removing a file that was created by [mistake](https://github.com/airbytehq/airbyte/pull/22938#issuecomment-1503948164)

**Pull request merged**